android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -198,6 +201,20 @@ public void getIpAddress(Promise p) {
     String ipAddress = Formatter.formatIpAddress(getWifiInfo().getIpAddress());
     p.resolve(ipAddress);
   }
+ 
+  @ReactMethod
+  public void getCameraPresence(Promise p) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      CameraManager manager=(CameraManager)getReactApplicationContext().getSystemService(Context.CAMERA_SERVICE);
+      try {
+        p.resolve(manager.getCameraIdList().length > 0);
+      } catch (CameraAccessException e) {
+        p.reject(e);
```

Comment:
The API doc you added has the .then() case but does not have a .catch() or indicate it could reject with an exception. Could you add a .catch() that says something like // CameraAccessException or similar ?

Reply:
done, but.... how do you think about  "always-resolves" and in this line implement samthing like this   
```
  } catch (CameraAccessException e) {
        p.resolve(undefined);
```

and type 
Promise<?boolean>

Reply:
Good question - I like it like this personally. An exception is an exception to me. This is a place of opinions though - everyone can have a different one but in the end your code will have to have some sort of branching logic to handle it, whichever style. So I'd leave it - you have this code working, you are using it in an app - that deployed experience is more valuable to me than switching a style

Reply:
=) yes we use this function, but don't catch error)))

Reply:
if reject camera permission, this exception does not throw

Reply:
interesting - it's documented as doing it so ? I dunno - the code seems obviously correct in reading. I'm running the branch now to test it, if it works, I still think this is the right sort of try/catch for this api...